SwiftUIHomeworks/SwiftUIHomeworks/Views/MainView.swift

```diff
@@ -0,0 +1,215 @@
+    }
+  }
+  
+  private var songImage: some View {
+    Image(playerViewModel.getCoverName())
+      .resizable()
+      .frame(maxWidth: UIScreen.main.bounds.width - Constants.oneHundredValue, maxHeight: UIScreen.main.bounds.height / Constants.threeValue)
+      .aspectRatio(contentMode: .fill)
+      .cornerRadius(Constants.cornerRadiusValue)
+  }
+  
+  private var artistText: some View {
+    Text(playerViewModel.getArtist())
+      .font(.title2)
+      .bold()
+      .padding()
+  }
+  
+  private var nameText: some View {
+    Text(playerViewModel.getName())
+      .padding()
+  }
+  
+  private var playerSlider: some View {
+    Slider(value: Binding(get: {
+      playerViewModel.durationSong
+    }, set: { newValue in
+      playerViewModel.durationSong = newValue
+      playerViewModel.setSongDuration(durationValue: playerViewModel.durationSong)
+    }), in: Constants.zeroValue...playerViewModel.maxDuration)
+  }
+  
+  private var currentTimeText: some View {
+    Text(playerViewModel.getCurrentTime())
+      .foregroundColor(.gray)
+  }
+  
+  private var durationText: some View {
+    Text(playerViewModel.getDuration())
+      .foregroundColor(.gray)
+  }
+  
+  private var sharingButton: some View {
+    Button {
+      self.isShareAlertShown = true
+    } label: {
+      Image(systemName: Constants.shareImageName)
+        .resizable()
+        .foregroundColor(.gray)
+        .aspectRatio(contentMode: .fill)
+        .frame(width: Constants.toolWidthValue, height: Constants.toolHeightValue)
+    }.alert(isPresented: $isShareAlertShown) {
+      Alert(title: Text(Constants.shareAlertTitle),
+            primaryButton: .destructive(Text(Constants.noTitle)),
+            secondaryButton: .default(Text(Constants.yesTitle)))
+    }
+  }
+  
+  private var saveButton: some View {
+    Button {
+      self.isSaveActionSheetShown = true
+    } label: {
+      Image(systemName: Constants.saveImageName)
+        .resizable()
+        .foregroundColor(.gray)
+        .aspectRatio(contentMode: .fill)
+        .frame(width: Constants.toolWidthValue, height: Constants.toolHeightValue)
+    }.actionSheet(isPresented: $isSaveActionSheetShown, content: {
+      ActionSheet(title: Text("\(playerViewModel.getName()) \(Constants.saveActionSheetText)"))
+    })
+  }
+  
+  private var linearGradientView: some View {
+    LinearGradient(gradient: gradient, startPoint: .top, endPoint: .bottom)
+  }
+  
+  private var gradient: Gradient {
+    Gradient(colors: [.clear, .secondary, Color(.systemGray4), Color(.systemGray3), Color(.systemGray2)])
+  }
+  
+  // MARK: - Private methods
+  
+  private func backgroundCoverView(coverName: String) -> (some View)? {
+    ZStack {
+      Image(coverName)
+        .resizable()
+      Rectangle()
+        .foregroundColor(.clear)
+        .background(linearGradientView)
+    }
+  }
+}
+
+struct MainView_Previews: PreviewProvider {
```

Comment:
```suggestion
struct MainView_Previews: PreviewProvider {
```
я хз, но тут, вроде как, тоже нужна дока...